zkvm/tests/zkvm.rs

```diff
@@ -39,71 +39,56 @@ fn test_helper(program: Vec<Instruction>) -> Result<TxID, VMError> {
 }
 
 fn issue_helper(
+    program: &mut Program,
     qty: u64,
     flv: Scalar,
     issuance_pred: Predicate,
     nonce_pred: Predicate,
-) -> Vec<Instruction> {
-    vec![
-        Instruction::Push(
-            Commitment::from(CommitmentWitness {
-                value: qty.into(),
-                blinding: Scalar::from(1u64),
-            })
-            .into(),
-        ), // stack: qty
-        Instruction::Var, // stack: qty-var
-        Instruction::Push(Commitment::from(CommitmentWitness::unblinded(flv)).into()), // stack: qty-var, flv
-        Instruction::Var,                        // stack: qty-var, flv-var
-        Instruction::Push(issuance_pred.into()), // stack: qty-var, flv-var, issuance-pred
-        Instruction::Issue,                      // stack: issue-contract
-        Instruction::Push(nonce_pred.into()),    // stack: issue-contract, nonce-pred
-        Instruction::Nonce,                      // stack: issue-contract, nonce-contract
-        Instruction::Signtx,                     // stack: issue-contract
-        Instruction::Signtx,                     // stack: issued-value
-    ]
+) {
+    program
+        .push(Commitment::from(CommitmentWitness {
+            value: qty.into(),
+            blinding: Scalar::from(1u64),
+        })) // stack: qty
+        .var() // stack: qty-var
+        .push(Commitment::from(CommitmentWitness::unblinded(flv))) // stack: qty-var, flv
```

Comment:
suggestion: compute `flv` from `issuance_pred` using `Value::issue_flavor()`, so we don't have to pass it by argument at all.

Reply:
Ew not sure I understand, `Value::issue_flavor()` gives a Scalar?

Reply:
Yes. It's the definition of a flavor based on the issuance predicate.

Spec: https://github.com/interstellar/slingshot/blob/main/zkvm/spec/ZkVM.md#issue

<img width="920" alt="image" src="https://user-images.githubusercontent.com/698/53364447-a94b3f00-38f3-11e9-87f9-74bb2fede170.png">